### PR TITLE
Improve storage doc

### PR DESCRIPTION
**Pull request by thiolliere**
* specification of how keys are inserted into storages are moved into generator docs
* decl_storage is more small and link to generator doc when necessary

# Critics

* maybe we should make the specification compiling, but all of this is tested in srml/support/test/test/final_keys.rs

### CODE REVIEW

File: srml/support/procedural/src/lib.rs [hunk 1]
```diff
@@ -72,46 +78,38 @@ use proc_macro::TokenStream;
...
 ///   [`Hashable`](../srml_support/trait.Hashable.html) trait.
 ///
-///   `hasher($hash)` is optional and its default is `blake2_256`.
-///
-///   /!\ Be careful with each key in the map that is inserted in the trie
-///   `$hash(module_name ++ " " ++ storage_name ++ encoding(key))`.
-///   If the keys are not trusted (e.g. can be set by a user), a cryptographic `hasher` such as
-///   `blake2_256` must be used. Otherwise, other values in storage can be compromised.
+///   `hasher($hash)` is optional and its default is `blake2_256`. Using another hasher must be
+///   made with care, see generator documentation.
```
Comment (thiolliere) [inline srml/support/procedural/src/lib.rs]: is this correct english ?

File: srml/support/procedural/src/lib.rs [hunk 2]
```diff
@@ -44,12 +44,18 @@ use proc_macro::TokenStream;
 /// with `Store` a (pub) trait generated associating each storage item to the `Module` and
 /// `as Example` setting the prefix used for storage items of this module. `Example` must be unique:
 /// another module with the same name and the same inner storage item name will conflict.
+/// `Example` is called the module prefix.
+///
+/// note: For instantiable modules the module prefix is prepend with instance
```
Comment (kianenigma) [inline srml/support/procedural/src/lib.rs]:
```suggestion
/// note: For instantiable modules the module prefix is prepended with instance
```

File: srml/support/procedural/src/lib.rs [hunk 3]
```diff
@@ -44,12 +44,18 @@ use proc_macro::TokenStream;
...
+/// prefix. Instance prefix is "" for default instance and "Instance$n" for instance number $n.
+/// Thus instance 3 of module Example has a module prefix of `Instance3Example`
```
Comment (kianenigma) [inline srml/support/procedural/src/lib.rs]:
```suggestion
/// Thus, instance 3 of module Example has a module prefix of `Instance3Example`
```

File: srml/support/procedural/src/lib.rs [hunk 4]
```diff
@@ -72,46 +78,38 @@ use proc_macro::TokenStream;
 ///   `$hash` representing a choice of hashing algorithms available in the
 ///   [`Hashable`](../srml_support/trait.Hashable.html) trait.
 ///
-///   `hasher($hash)` is optional and its default is `blake2_256`.
-///
-///   /!\ Be careful with each key in the map that is inserted in the trie
-///   `$hash(module_name ++ " " ++ storage_name ++ encoding(key))`.
-///   If the keys are not trusted (e.g. can be set by a user), a cryptographic `hasher` such as
-///   `blake2_256` must be used. Otherwise, other values in storage can be compromised.
+///   `hasher($hash)` is optional and its default is `blake2_256`. Using another hasher must be
```
Comment (kianenigma) [inline srml/support/procedural/src/lib.rs]:
```suggestion
///   `hasher($hash)` is optional and its default is `blake2_256`. One should use another hasher
```

File: srml/support/procedural/src/lib.rs (hunk 1 above)
Comment (kianenigma) [inline srml/support/procedural/src/lib.rs]:
```suggestion
///  with care, see generator documentation.
```

File: srml/support/src/storage/generator/linked_map.rs [hunk 5]
```diff
@@ -23,8 +23,26 @@ use rstd::{
...
-/// For each key value is stored at `Hasher(prefix ++ key)` along with a linkage used for
-/// enumeration.
+/// # Mapping of keys to a storage path
+///
+/// The key for the head of the map is stored at one fixed path:
+/// ```nocompile
+/// Hasher(head_key)
+/// ```
+///
+/// For each key the value is stored appended by a [`Linkage`](struct.Linkage.html) (which hold
```
Comment (kianenigma) [inline srml/support/src/storage/generator/linked_map.rs]:
```suggestion
/// For each key the value stored is appended by a [`Linkage`](struct.Linkage.html) (which hold
```

**Pull request merged**